File: tests_wraps/src/main/java/com/github/klee0kai/test/car/model/CarInjectLists.java
package com.github.klee0kai.test.car.model;

import javax.inject.Inject;
import java.util.List;

public class CarInjectLists {


    @Inject
    public List<Bumper> bumpers;

    @Inject
    public List<Wheel> wheels;

    @Inject
    public List<Window> windows;

    public List<Bumper> bumpersMethodFrom;

    public List<Wheel> wheelsMethodFrom;

    public List<Window> windowsMethodFrom;

    @Inject
    public void init(List<Bumper> bumpers, List<Wheel> wheels, List<Window> windows) {
        bumpersMethodFrom = bumpers;
        wheelsMethodFrom = wheels;
        windowsMethodFrom = windows;
    }


}

File: tests_wraps_kotlin/src/main/kotlin/com/github/klee0kai/test/car/model/CarInjectLists.kt
package com.github.klee0kai.test.car.model

import com.github.klee0kai.stone.annotations.qualifier.IgnoreQualifier
import javax.inject.Inject

class CarInjectLists {

    @Inject
    @IgnoreQualifier
    var bumpers: List<Bumper>? = null

    @Inject
    @IgnoreQualifier
    var wheels: List<Wheel>? = null

    @Inject
    @IgnoreQualifier
    var windows: List<Window>? = null


    var bumpersMethodFrom: List<Bumper>? = null
    var wheelsMethodFrom: List<Wheel>? = null
    var windowsMethodFrom: List<Window>? = null

    @Inject
    fun init(
        @IgnoreQualifier
        bumpers: List<Bumper>?,
        @IgnoreQualifier
        wheels: List<Wheel>?,
        @IgnoreQualifier
        windows: List<Window>?,
    ) {
        bumpersMethodFrom = bumpers
        wheelsMethodFrom = wheels
        windowsMethodFrom = windows
    }
}
